clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1437,20 +1670,23 @@ void Sema::ConstructOpenCLKernel(FunctionDecl *KernelCallerFunc,
                                                        : CalculatedName);
 
   SyclKernelFieldChecker checker(*this);
-  SyclKernelDeclCreator kernel_decl(*this, checker, KernelName,
```

Comment:
So instead of this, we should be making the VisitRecordFields not continue with a field once it finds an error.  visit function should just return true/false to determine whether it needs to continue.

Reply:
The visitor loop is currently written like this: 
for-each-field { check, do kernel caller declaration, do kernel body creation, do integration header creation}. 
Even after the checker finds a bad field the kernel declaration, body creation and integration header creation are done, leaving that code exposed to possible trip-ups caused by unexpected input. An invalid field should not be processed any further.

This PR modifies the loop to this: 
for-each-field { check }
if (valid) for-each-field { do kernel caller declaration, do kernel body creation, do integration header creation}

If you are proposing to keep the visit loop as is, but for each handler to check if the field is OK before proceeding, the changes to the code will be wide-spread. Each handler will need to return a bool because each handler function requires a uniform signature to fit into the Visitor calling pattern. That is 9 handlers x 4 processors = 36 function signatures.